magenta/models/rl_tuner/rl_tuner.py

```diff
@@ -0,0 +1,2052 @@
+        learning algorithms
+      note_rnn_checkpoint_dir: The directory from which the internal NoteRNNLoader
+        will load its checkpointed LSTM.
+      backup_checkpoint_file: A checkpoint file to use in case one cannot be
+        found in the note_rnn_checkpoint_dir.
+      note_rnn_type: If 'default', will use the basic LSTM described in the 
+        research paper. If 'basic_rnn', will assume the checkpoint is from a
+        Magenta basic_rnn model.
+      note_rnn_hparams: A tf.HParams object which defines the hyper parameters
+        used to train the MelodyRNN model that will be loaded from a checkpoint.
+      num_notes_in_melody: The length of a composition of the model
+      midi_primer: A midi file that can be used to prime the model if
+        priming_mode is set to 'single_midi'.
+      input_size: the size of the one-hot vector encoding a note that is input
+        to the model.
+      num_actions: The size of the one-hot vector encoding a note that is
+        output by the model.
+      save_name: Name the model will use to save checkpoints.
+      output_every_nth: How many training steps before the model will print
+        an output saying the cumulative reward, and save a checkpoint.
+      training_file_list: A list of paths to tfrecord files containing melody 
+        training data. This is necessary to use the 'random_midi' priming mode.
+      summary_writer: A tf.train.SummaryWriter used to log metrics.
+      initialize_immediately: if True, the class will instantiate its component
+        MelodyRNN networks and build the graph in the constructor.
+    """
+    # Make graph.
+    self.graph = tf.Graph()
+
+    with self.graph.as_default():
+      # Memorize arguments.
+      self.input_size = input_size
+      self.num_actions = num_actions
+      self.output_every_nth = output_every_nth
+      self.output_dir = output_dir
+      self.save_path = os.path.join(output_dir, save_name)
+      self.reward_scaler = reward_scaler
+      self.reward_mode = reward_mode
+      self.exploration_mode = exploration_mode
+      self.num_notes_in_melody = num_notes_in_melody
+      self.stochastic_observations = stochastic_observations
+      self.algorithm = algorithm
+      self.priming_mode = priming_mode
+      self.midi_primer = midi_primer
+      self.note_rnn_checkpoint_dir = note_rnn_checkpoint_dir
+      self.training_file_list = training_file_list
+      self.backup_checkpoint_file = backup_checkpoint_file
+      self.note_rnn_hparams = note_rnn_hparams
+      self.note_rnn_type = note_rnn_type
+
+      if priming_mode == 'single_midi' and midi_primer is None:
+        tf.logging.fatal('A midi primer file is required when using'
+                         'the single_midi priming mode.')
+
+      if note_rnn_checkpoint_dir is None or note_rnn_checkpoint_dir == '':
+        print 'Retrieving checkpoint of Note RNN from Google cloud storage.'
```

Comment:
Instead of "Google cloud storage", I'd just say "Magenta download server" or something.